algorithms_task_2/student/include/impl.h

```diff
@@ -9,4 +10,16 @@
  *
  * @return true if source starts from a prefix
  */
-bool startsWith(const std::string& source, const std::string& prefix);
+bool startsWith(const std::string& source, const std::string& prefix){
+    return std::equal(prefix.begin(),prefix.end(),source.begin()); 
+    // Complexity: prefix.begin()-prefix.end() applications of the predicate
+
+    // Alternatives:
+    // 1) return source.find(prefix)==0; 
```

Comment:
What if prefix will be at the end?

Reply:
Yes, I know if prefix is at the end or also if prefix is not found, the code will take linear time, so it will be ineffective. I think it would be possible to check the first character before calling the find method in order to speed up the code. Anyway this alternative is less efficient and more complicated than the std::equal